sdk/ml/azure-ai-ml/azure/ai/ml/operations/_datastore_operations.py

```diff
@@ -220,3 +227,84 @@ def create_or_update(self, datastore: Datastore) -> Datastore:
                 log_and_raise_error(ex)
             else:
                 raise ex
+
+    @monitor_with_activity(logger, "Datastore.Mount", ActivityType.PUBLICAPI)
+    @experimental
+    def mount(
+        self,
+        path: str,
+        mount_point: str = None,
+        mode: str = "ro_mount",
+        debug: bool = False,
+        persistent: bool = False,
+        **_kwargs,
+    ) -> None:
+        """Mount a datastore to a local path.
+
+        :param path: The data store path to mount, in the form of `<name>` or `azureml://datastores/<name>`.
+        :type path: str
+        :param mount_point: A local path used as mount point.
+        :type mount_point: str
+        :param mode: Mount mode, either `ro_mount` (read-only) or `rw_mount` (read-write).
+        :type mode: str
+        :param debug: Whether to enable verbose logging.
+        :type debug: bool
+        :param persistent: Whether to persist the mount after reboot. Applies only when running on Compute Instance.
```

Comment:
nit: Explain clearly what the requirement for persistence is.

```suggestion
        :param persistent: Whether to persist the mount after reboot. Applies only when running on Compute Instance,
                where the 'CI_NAME' environment variable is set."
```